src/accelerate/utils/dataclasses.py

```diff
@@ -1810,6 +1811,31 @@ def set_mixed_precision(self, mixed_precision, buffer_autocast=False, override=F
             self.mixed_precision_policy = MixedPrecision(**self.mixed_precision_policy)
 
 
+@dataclass
+class TorchTensorParallelPlugin:
+    """
+    This plugin is used to enable tensor parallelism using PyTorch 2.0.
```

Comment:
I think a reference to further information on TP should be added here. Also:

```suggestion
    This plugin is used to enable tensor parallelism using PyTorch >= 2.0.
```